### ISSUE TITLE
Tables & Overlapping entities

### ISSUE THREAD

**Issue by SamyPesse** [facebook/draft-js#212]
First of all, thank you for your awesome work :beers: 

I'm currently switching from `CodeMirror + Custom widgets` to `draft-js` for our [GitBook Editor](https://www.gitbook.com/editor).
I've started by implementing a library, [draft-markup](https://github.com/GitbookIO/draft-markup) to parse `Markdown` to `RawContentState`, and `RawContentState` to `Markdown` (or other markup syntaxes). You can see checkout an example at http://gitbookio.github.io/draft-markup/

I successfully implemented the markdown syntax using the rule of our markdown parser: link, emphasis, image, code, etc. But I'm facing issue on the implementation of tables.

My original idea was to parse a table like:

```markdown
| Col 1 | Col 2 |
| ----- | ----- |
| Cell 1.1 | Cell 1.2 |
| Cell 2.1 | Cell 2.2 |
```

into one `TABLE` block with 4 types of entities in it: `TABLE_HEADER`, `TABLE_BODY`, `TABLE_ROW` and `TABLE_CELL`.

But It seems that draft-js doesn't support overlapping entities (since `TABLE_CELL` is included inside `TABLE_ROW`, which is included inside `TABLE_BODY`).

Basically, here is the `RawContentState` when parsing the previous table: https://gist.github.com/SamyPesse/589225b9d6728bf1db6f

And here is the dom generated by `draft-js`:

![screen shot 2016-03-15 at 09 31 09](https://cloud.githubusercontent.com/assets/845425/13772259/6e26a07e-ea93-11e5-85ef-8dc4dc224e52.png)

I understand that overlapping entities are not a good idea and should be avoided,but  what would be the advised method for editing table with `draft-js` ?

**Comment by hellendag** [comment]
My suggestion for tables is to use an entity and a custom block renderer with an input mechanism, much like the TeX example.
